Tests/MLXTests/ModuleTests.swift

```diff
@@ -698,4 +698,31 @@ class ModuleTests: XCTestCase {
             XCTFail("should be a .value(array)")
         }
     }
+
+    func testUpdateModulesNil() {
+
+        class PatchMerger: Module {
+            @ModuleInfo var mlp: (Linear, GELU, Linear)
+
+            override init() {
+                mlp = (
+                    Linear(128, 128),
+                    GELU(),
+                    Linear(128, 128)
+                )
+            }
+        }
+
+        let pm = PatchMerger()
+
+        // this will fail because the quantize will
+        // do an update with (QL, nil, QL) and it can't
+        // replace the GELU with nil.  With the fix
+        // this works and forward the GELU
```

Comment:
Tiny nit.. this comment is useful for the PR, but I think won't age too well in the code. It references that it will fail which it no longer does and references a fix but without saying much else.

Reply:
Good point -- this was intended to describe the situation that would fail (and it is testing for) but let me rephrase it a bit.